src/Aquifer.API/Endpoints/Resources/Content/ResourceStatusHelpers.cs

```diff
@@ -0,0 +1,130 @@
+        {
+            await historyService.AddSnapshotHistoryAsync(draftVersion, draftVersion.AssignedUserId, originalStatus, ct);
+        }
+
+        if (draftVersion.AssignedUserId != assignedUserId)
+        {
+            draftVersion.AssignedUserId = assignedUserId;
+            await historyService.AddAssignedUserHistoryAsync(draftVersion, assignedUserId, user.Id, ct);
+        }
+
+        if (draftVersion.ResourceContent.Status != originalStatus)
+        {
+            await historyService.AddStatusHistoryAsync(draftVersion, draftVersion.ResourceContent.Status, user.Id, ct);
+        }
+    }
+
+    public static void SetAssignedReviewerUserId(int? assignedReviewerUserId, ResourceContentVersionEntity draftVersion)
+    {
+        if (assignedReviewerUserId is not null)
+        {
+            draftVersion.AssignedReviewerUserId = assignedReviewerUserId;
+        }
+    }
+
+    public static void ValidateAllowedToAssign<TRequest>(AssignmentPermissions permissions,
+        ResourceContentVersionEntity draftVersion,
+        UserEntity user)
+    {
+        var currentUserIsAssigned = draftVersion.AssignedUserId == user.Id;
+        var assignedUserIsInCompany = draftVersion.AssignedUser?.CompanyId == user.CompanyId;
+
+        var allowedToAssign = (permissions.HasAssignOverridePermission &&
+                              (assignedUserIsInCompany || permissions.HasAssignOutsideCompanyPermission)) ||
+                              currentUserIsAssigned;
+
+        if (!allowedToAssign)
+        {
+            ValidationContext<TRequest>.Instance.ThrowError($"Unable to assign user for id {draftVersion.ResourceContentId}");
+        }
+    }
+
+    public static async Task<List<ResourceContentVersionEntity>> GetDraftVersions<TRequest>(int[] contentIds,
+        List<ResourceContentStatus> allowedStatuses, AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        var draftVersions = await dbContext.ResourceContentVersions
+            .AsTracking()
+            .Where(rcv => contentIds.Contains(rcv.ResourceContentId) && allowedStatuses.Contains(rcv.ResourceContent.Status) && rcv.IsDraft)
+            .Include(rcv => rcv.ResourceContent)
+            .Include(rcv => rcv.AssignedUser).Include(rcv => rcv.ResourceContentVersionSnapshots)
+            .ToListAsync(ct);
+
+        if (draftVersions.Count != contentIds.Length)
+        {
+            ValidationContext<TRequest>.Instance.ThrowError("One or more resources not found or not in draft status");
+        }
+
+        return draftVersions;
+    }
+
+    public static async Task ValidateReviewerAndAssignedUser<TRequest>(int assignedUserId, int? assignedReviewerUserId,
+        AquiferDbContext dbContext, UserEntity user, AssignmentPermissions permissions,
+        CancellationToken ct,
+        UserEntity? userToAssign = null)
+    {
+        var assignedUser = userToAssign ?? await dbContext.Users
```

Comment:
I don't think we get anything out of this extra variable now.
```suggestion
        userToAssign ??= await dbContext.Users
            .AsTracking()
            .SingleOrDefaultAsync(u => u.Id == assignedUserId && u.Enabled, ct);
```